Allow library external functions to be attached with using for (#13855)
ethereum/solidity: Changelog.md

 * SMTChecker: Fix internal error on chain assignments using static fully specified state variables.
 * SMTChecker: Fix internal error when using user defined types as mapping indices or struct members.
 * SMTChecker: Fix internal error when deleting struct member of function type.

>  * TypeChecker: Also allow external library functions in ``using for {...}``.

Reviewer: Shouldn't it be `using {...} for` instead?

Anyway, I'd just say `using for`:
```suggestion
 * TypeChecker: Also allow external library functions in ``using for``.
```

>  * TypeChecker: Fix external library functions not being allowed when individually specified in ``using for``.

Reviewer 2: ```suggestion
 * TypeChecker: Also allow external library functions in ``using for {...}``.
```
may be easier to read.

 * SMTChecker: Fix internal error when using user defined types as mapping indices or struct members.
 * SMTChecker: Fix internal error when deleting struct member of function type.
 * TypeChecker: Fix bug where private library functions could be attached with ``using for`` outside of their declaration scope.

>  * TypeChecker: Also allow external library functions in ``using for``.

Reviewer: Let's keep these ordered alphabetically.

Author: Right, forgot about that.
